### ISSUE TITLE
MooseDocs misses a few build failures

### ISSUE THREAD

**Issue by permcody** [idaholab/moose#9908]
### Description of the enhancement or error report

MOOSE build sometimes misses build failures (e.g. reports success when there are failures). See this job: https://moosebuild.inl.gov/job/84609/

### Rationale for the enhancement or information for reproducing the error

Unknown, @gardnerru can you shed some light on what the problem was and how it was caused/fixed?

### Identified impact
(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)

This bug allows us to check in broken docs.

Tag @aeslaughter

**Comment by gardnerru** [comment]
Bison MR 921 is the example where I saw this. There were actually a couple problems, one was caused by not updating before committing, which made the "sync to remote" test fail. Unfortunately there was also duplicate bibliography entries, which caused a python fail as well so no docs were made, but the "build docs" test showed green. Once the user updated and re-pushed the tests showed all green but the docs did not build due to the duplicates. 

<img width="1016" alt="screen shot 2017-09-23 at 11 07 55 am" src="https://user-images.githubusercontent.com/6787866/30775382-960f43fc-a04f-11e7-956d-351400821d4c.png">
